### PR TITLE
[ENGA3-251] added pipeline to deploy on staging

### PR DESCRIPTION
**Pull request by ajzkk**
#### 1. Objective

Added pipeline to deploy on staging

**Related information**:
- [ENGA3-251](https://opn-ooo.atlassian.net/browse/ENGA3-251)
- [Buildkite pipeline](https://buildkite.com/omise/magento-docker)
- [Buildkite REST API](https://buildkite.com/docs/apis/rest-api/builds#create-a-build)

#### 2. Description of change

Once the branched merged into master, this pipeline will run and call buildkite pipeline API to deploy on staging.

#### 5. Priority of change

Normal

#### 6. Additional Notes

### CODE REVIEW

File: .github/workflows/deploy-staging.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: Deploy to Staging
+on:
+  push:
+    branches: [ 'master' ]
```
Comment (kan98) [inline .github/workflows/deploy-staging.yml:4]: should we also deploy from staging branch? so you can test things before you merge to master 🤔

File: .github/workflows/deploy-staging.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+jobs:
+  static:
+    name: Deploy to Staging
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      
+      - name: Trigger Buildkite Pipeline & Deploy On Staging
+        run: sh .github/workflows/trigger-buildkite-pipeline.sh ${{ secrets.BUILDKITE_API_ACCESS_TOKEN }}
```
Comment (kan98) [inline .github/workflows/deploy-staging.yml:14]:
```suggestion
        run: sh .github/workflows/trigger-buildkite-pipeline.sh ${{ secrets.BUILDKITE_API_ACCESS_TOKEN }}

```
EOF

File: .github/workflows/trigger-buildkite-pipeline.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+PAYLOAD='{"commit": "HEAD", "branch": "master", "message": "Triggered From omise-magento Plugin"}'
+curl -H "Authorization: Bearer $1" -X POST "https://api.buildkite.com/v2/organizations/omise/pipelines/magento-docker/builds" -d "$PAYLOAD"
```
Comment (kan98) [inline .github/workflows/trigger-buildkite-pipeline.sh:4]:
```suggestion
curl -H "Authorization: Bearer $1" -X POST "https://api.buildkite.com/v2/organizations/omise/pipelines/magento-docker/builds" -d "$PAYLOAD"

```
EOF

File: .github/workflows/deploy-staging.yml (hunk 1 above)
Comment (ajzkk) [reply .github/workflows/deploy-staging.yml:4]: Agreed. @davidhanh @aashishgurung. WDYT?

Comment (ajzkk) [reply .github/workflows/deploy-staging.yml:4]: @davidhanh suggest to go with this new flow [Branching Strategy](https://opn-ooo.atlassian.net/wiki/spaces/D/pages/576754440/Branching+Strategy+-+Scaled+Trunk+Based+Development).

cc: @kan98 @aashishgurung

Comment (aashishgurung) [reply .github/workflows/deploy-staging.yml:4]: @kan98 @ajzkk Once our PR is merged and deployed to the staging, QA will do the testing in the staging and reopens the ticket if they find any issue. So, I don't think we need staging branch. Plus current process complies with the new branching strategy.

**Review by aashishgurung — APPROVED** [review]

File: .github/workflows/deploy-staging.yml (hunk 1 above)
Comment (kan98) [reply .github/workflows/deploy-staging.yml:4]: yea that's up to you guys how it'll work best for you. I agree that if you're always gonna test on master first then this is fine. I'll approve the PR

**Review by kan98 — APPROVED** [review]

**Review by new4762 — APPROVED** [review]

**Pull request merged**